docs/reference/docs/concurrency-control.asciidoc

```diff
@@ -0,0 +1,111 @@
+[[optimistic-concurrency-control]]
+== Optimistic concurrency control
+
+Elasticsearch is distributed. When documents are created, updated, or deleted,
+the new version of the document has to be replicated to other nodes in the cluster. 
+Elasticsearch is also asynchronous and concurrent, meaning that these replication
+requests are sent in parallel, and may arrive at their destination out of sequence. 
+Elasticsearch needs a way of ensuring that an older version of a document never 
+overwrites a newer version.
+
+To ensure this, every operation performed to a document is assigned a sequence number
```

Comment:
To ensure an older version of a document doesn't overwrite a newer version, every operation performed is assigned a sequence number ...